sopel/irc.py

```diff
@@ -419,83 +408,55 @@ def found_terminator(self):
-                    if line.startswith('File "'):
-                        report.append(line[0].lower() + line[1:])
-                        break
-                else:
-                    report.append('source unknown')
-
-                signature = '%s (%s)' % (report[0], report[1])
-                # TODO: make not hardcoded
-                log_filename = os.path.join(self.config.core.logdir, self.config.basename + '.exceptions.log')
-                with codecs.open(log_filename, 'a', encoding='utf-8') as logfile:
-                    logfile.write('Signature: %s\n' % signature)
-                    if trigger:
-                        logfile.write('from {} at {}. Message was: {}\n'.format(
-                            trigger.nick, str(datetime.now()), trigger.group(0)))
-                    logfile.write(trace)
-                    logfile.write(
-                        '----------------------------------------\n\n'
-                    )
-            except Exception as e:
-                stderr("Could not save full traceback!")
-                LOGGER.error("Could not save traceback from %s to file: %s", trigger.sender, str(e))
-
-            if trigger and self.config.core.reply_errors and trigger.sender is not None:
-                self.say(signature, trigger.sender)
-            if trigger:
-                LOGGER.error('Exception from {}: {} ({})'.format(trigger.sender, str(signature), trigger.raw))
-        except Exception as e:
-            if trigger and self.config.core.reply_errors and trigger.sender is not None:
-                self.say("Got an error.", trigger.sender)
-            if trigger:
-                LOGGER.error('Exception from {}: {} ({})'.format(trigger.sender, str(e), trigger.raw))
+        message = 'Unexpected error'
+        if exception:
+            message = '{} ({})'.format(message, exception)
+
+        if trigger:
+            message = '{} from {} at {}. Message was: {}'.format(
+                message, trigger.nick, str(datetime.now()), trigger.group(0)
+            )
+
+        LOGGER.exception(message)
+
+        if trigger and self.config.core.reply_errors and trigger.sender is not None:
+            self.say(message, trigger.sender)
 
     def handle_error(self):
         """Handle any uncaptured error in the core.
 
-        Overrides asyncore's handle_error.
-
+        This method is an override of :meth:`asyncore.dispatcher.handle_error`,
+        the :class:`asynchat.async_chat` being a subclass of
+        :class:`asyncore.dispatcher`.
         """
-        trace = traceback.format_exc()
-        stderr(trace)
-        LOGGER.error('Fatal error in core, please review exception log')
-        # TODO: make not hardcoded
-        logfile = codecs.open(
-            os.path.join(self.config.core.logdir, self.config.basename + '.exceptions.log'),
-            'a',
-            encoding='utf-8'
+        LOGGER.error('Fatal error in core, please review exceptions logs.')
```

Comment:
```suggestion
        LOGGER.error('Fatal error in core; please review exception log.')
```

May or may not be useful enough to put the actual log filename here, if we can get it.

Reply:
Hm. That's an idea. I'd suggest to improve that in another PR.